f5_openstack_agent/lbaasv2/drivers/bigip/test/test_network_service.py

```diff
@@ -0,0 +1,223 @@
+            'ip_version': 4,
+            'ipv6_address_mode': None,
+            'cidr': '172.32.254.0/24',
+            'id': '1189a88f-0fe7-4d0e-8549-b557d7f7e98b',
+            'subnetpool_id': None,
+            'name': 'subnet-new'}
+
+
+@pytest.fixture
+def network_service(rds_cache):
+    conf = mock.MagicMock()
+    conf.vlan_binding_driver = None
+    driver = mock.MagicMock()
+    driver.conf = conf
+    service = NetworkServiceBuilder(False, conf, driver)
+
+    # add a 'real' RD cache
+    service.rds_cache = rds_cache
+
+    # mock NetworkHelper.get_route_domain()
+    rd = namedtuple('RouteDomain', 'id')
+    rd.id = 1234
+    service.network_helper = mock.MagicMock()
+    service.network_helper.get_route_domain = mock.MagicMock(return_value=rd)
+
+    return service
+
+
+class TestNetworkServiceBuilder(object):
+    """Test assign route domain in NetworkServiceBuilder class"""
+
+    def test_get_neutron_net_short_name(self, network_service, network):
+        """Test function that creates network name.
+
+        """
+        # valid name
+        net_short_name = network_service.get_neutron_net_short_name(network)
+        assert net_short_name == 'vlan-608'
+
+        # invalid network type
+        with pytest.raises(f5_ex.InvalidNetworkType):
+            network['provider:network_type'] = ''
+            network_service.get_route_domain_from_cache(network)
+
+        # invalid segmentation ID
+        with pytest.raises(f5_ex.InvalidNetworkType):
+            network['provider:network_type'] = 'vlan'
+            network['provider:segmentation_id'] = ''
+            network_service.get_route_domain_from_cache(network)
+
+    def test_get_route_domain_from_cache(self, network_service, network):
+        # valid cache entries
+        rd = network_service.get_route_domain_from_cache(network)
+        assert rd == 2
+
+        network['provider:segmentation_id'] = 604
+        rd = network_service.get_route_domain_from_cache(network)
+        assert rd == 1
+
+        # vlan not in cache
+        with pytest.raises(f5_ex.RouteDomainCacheMiss) as excinfo:
+            network['provider:segmentation_id'] = 600
+            network_service.get_route_domain_from_cache(network)
+        assert 'vlan-600' in str(excinfo.value)
+
+        # empty cache
+        with pytest.raises(f5_ex.RouteDomainCacheMiss) as excinfo:
+            network['provider:segmentation_id'] = 606
+            network['provider:network_type'] = 'vlan'
+            network_service.rds_cache = {}
+            network_service.get_route_domain_from_cache(network)
+        assert 'vlan-606' in str(excinfo.value)
+
+        # invalid network data
+        with pytest.raises(f5_ex.InvalidNetworkType):
+            network['provider:segmentation_id'] = 604
+            network['provider:network_type'] = ''
+            network_service.get_route_domain_from_cache(network)
+
+    def test_assign_route_domain(self, network_service, network, subnet):
```

Comment:
An additional test that is needed is to show that we log the message we expect to when we call assign_route_domain and we get a RouteDomainCacheMiss. The other test is just to ensure we actually want to raise f5_ex.InvalidNetworkType up when we call assign_route_domain and a bad network type or seg id has been given. That would raise in get_neutron_net_short_name, then go up to get_route_domain_from_cache, and finally up to assign_route_domain.